fastiron-stats/src/structures.rs

```diff
@@ -0,0 +1,138 @@
+//! Modelling code
+//!
+//! This module contains all structures used to model the data and
+//! structure the computations.
+
+use std::{iter::zip, ops::Index};
+
+/// Number of fields presented in the tallies report produced
+/// by `fastiron`.
+pub const N_TALLIED_DATA: usize = 17;
+
+/// Enum used to represent & map tallied data and their indexes.
+#[derive(Debug)]
+pub enum TalliedData {
+    Cycle = 0,
+    Start = 1,
+    Source = 2,
+    Rr = 3,
+    Split = 4,
+    Absorb = 5,
+    Scatter = 6,
+    Fission = 7,
+    Produce = 8,
+    Collision = 9,
+    Escape = 10,
+    Census = 11,
+    NumSeg = 12,
+    ScalarFlux = 13,
+    PopulationControl = 14,
+    CycleTracking = 15,
+    CycleSync = 16,
+}
+
```

Comment:
Perhaps this should be provided by `fastiron` to be sure that all is always correctly synchronized.